applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/harmonic_analysis_strategy.hpp

```diff
@@ -0,0 +1,743 @@
+        const auto rank = rModelPart.GetCommunicator().MyPID();
+
+        if (BaseType::GetEchoLevel() > 2 && rank == 0)
+            std::cout << "Entering Initialize() of HarmonicAnalysisStrategy." << std::endl;
+
+        this->Check();
+
+        auto& pScheme = this->pGetScheme();
+
+        if (pScheme->SchemeIsInitialized() == false)
+            pScheme->Initialize(rModelPart);
+
+        if (pScheme->ElementsAreInitialized() == false)
+            pScheme->InitializeElements(rModelPart);
+
+        if (pScheme->ConditionsAreInitialized() == false)
+            pScheme->InitializeConditions(rModelPart);
+
+        if (BaseType::GetEchoLevel() > 2 && rank == 0)
+            std::cout << "Exiting Initialize() of HarmonicAnalysisStrategy." << std::endl;
+
+        // set up the system
+        auto& pBuilderAndSolver = this->pGetBuilderAndSolver();
+
+        // Reset solution dofs
+        boost::timer system_construction_time;
+        // Set up list of dofs
+        boost::timer setup_dofs_time;
+        pBuilderAndSolver->SetUpDofSet(pScheme, rModelPart);
+        if (BaseType::GetEchoLevel() > 0 && rank == 0)
+        {
+            std::cout << "setup_dofs_time : " << setup_dofs_time.elapsed() << std::endl;
+        }
+
+        // Set global equation ids
+        boost::timer setup_system_time;
+        pBuilderAndSolver->SetUpSystem(rModelPart);
+        if (BaseType::GetEchoLevel() > 0 && rank == 0)
+        {
+            std::cout << "setup_system_time : " << setup_system_time.elapsed() << std::endl;
+        }
+
+        // initialize the force vector; this does not change during the computation
+        auto& pForceVector = this->pGetForceVector();
+        auto& rForceVector = *pForceVector;
+        const unsigned int system_size = pBuilderAndSolver->GetEquationSystemSize();
+        
+        boost::timer force_vector_build_time;
+        if (rForceVector.size() != system_size)
+            rForceVector.resize(system_size, false);
+        rForceVector = ZeroVector( system_size );
+        pBuilderAndSolver->BuildRHS(pScheme,rModelPart,rForceVector);
+        
+        if (BaseType::GetEchoLevel() > 0 && rank == 0)
+        {
+            std::cout << "force_vector_build_time : " << force_vector_build_time.elapsed() << std::endl;
+        }
+
+        // initialize the modal matrix
+        auto& pModalMatrix = this->pGetModalMatrix();
+        auto& rModalMatrix = *pModalMatrix;
+        const std::size_t n_modes = rProcessInfo[EIGENVALUE_VECTOR].size();
+        if( rModalMatrix.size1() != system_size || rModalMatrix.size2() != n_modes )
+            rModalMatrix.resize( system_size, n_modes, false );
+        rModalMatrix = ZeroMatrix( system_size, n_modes );
+
+        boost::timer modal_matrix_build_time;
+        for( std::size_t i = 0; i < n_modes; ++i )
+        {
+            for( ModelPart::NodeIterator itNode = rModelPart.NodesBegin(); itNode != rModelPart.NodesEnd(); itNode++ )
```

Comment:
here (I think) you can use an `auto` loop
sth like `for (const auto & node : rModelPart.Nodes())`